xds/src/test/java/io/grpc/xds/ClientXdsClientDataTest.java

```diff
@@ -879,6 +887,105 @@ public void parseHttpFilter_unsupportedButOptional() {
     assertThat(ClientXdsClient.parseHttpFilter(httpFilter, filterRegistry, true)).isNull();
   }
 
+  private static class TestFilter implements io.grpc.xds.Filter,
+      io.grpc.xds.Filter.ClientInterceptorBuilder {
+    @Override
+    public String[] typeUrls() {
+      return new String[]{"test-url"};
+    }
+
+    @Override
+    public ConfigOrError<? extends FilterConfig> parseFilterConfig(Message rawProtoMessage) {
+      return ConfigOrError.fromConfig(new FilterConfig() {
```

Comment:
Echo back the protobuf message in the return value so that you can verify the config was actually propagated properly?